pyro/distributions/torch_distribution.py

```diff
@@ -258,6 +252,30 @@ def __init__(self, base_dist, sample_shape=torch.Size(), reinterpreted_batch_ndi
         batch_shape, event_shape = shape[:batch_dim], shape[batch_dim:]
         super(ReshapedDistribution, self).__init__(batch_shape, event_shape)
 
+    def expand(self, batch_shape):
+        batch_shape = torch.Size(batch_shape)
+        # Raise error if existing batch shape is being shrunk.
+        # e.g. (2, 4) -> (2, 1)
+        proposed_shape = broadcast_shape(self.batch_shape, batch_shape)
+        if tuple(reversed(proposed_shape)) > tuple(reversed(batch_shape)):
+            raise ValueError("Existing batch shape {} cannot be expanded "
+                             "to the new batch shape {}."
+                             .format(self.batch_shape, batch_shape))
+        # Adjust existing sample shape if possible.
+        base_dist = self.base_dist
+        base_batch_shape = batch_shape + self.event_shape[:self.reinterpreted_batch_ndims]
+        cut = len(base_batch_shape) - len(base_dist.batch_shape)
```

Comment:
Is this safe in cases when `cut < 0`?

Reply:
`cut` should not be less than 1. If it is, we will end up throwing a ValueError above because in that case the proposed batch shape will be less than the original batch shape. Let me modify the tests to assert that.